File: java/src/main/java/nl/craftsmen/workshops/reactivemeetup/domain/railway/TrainSimulationParameters.java
package nl.craftsmen.workshops.reactivemeetup.domain.railway;

public class TrainSimulationParameters {
	
	private String trainId;
	
	private double maxVelocity;
	
	private double acceleration;
	
	private double timeDilation;
	
	private int tickFrequency;

	public String getTrainId() {
		return trainId;
	}

	public TrainSimulationParameters setTrainId(String trainId) {
		this.trainId = trainId;
		return this;
	}

	public double getMaxVelocity() {
		return maxVelocity;
	}

	public TrainSimulationParameters setMaxVelocity(double maxVelocity) {
		this.maxVelocity = maxVelocity;
		return this;
	}

	public double getAcceleration() {
		return acceleration;
	}

	public TrainSimulationParameters setAcceleration(double acceleration) {
		this.acceleration = acceleration;
		return this;
	}

	public double getTimeDilation() {
		return timeDilation;
	}

	public TrainSimulationParameters setTimeDilation(double timeDilation) {
		this.timeDilation = timeDilation;
		return this;
	}

	public int getTickFrequency() {
		return tickFrequency;
	}

	public TrainSimulationParameters setTickFrequency(int tickFrequency) {
		this.tickFrequency = tickFrequency;
		return this;
	}
		
}

File: javascript/src/domain/railway/train-simulation-parameters.js
module.exports = class TrainSimulationParameters {

    constructor(tickFrequency, maxVelocity, acceleration, trainId, timeDilation) {
        this.tickFrequency = tickFrequency;
        this.maxVelocity = maxVelocity;
        this.acceleration = acceleration;
        this.trainId = trainId;
        this.timeDilation = timeDilation;
    }

    getTrainId() {
        return this.trainId;
    }

    getMaxVelocity() {
        return this.maxVelocity;
    }

    getAcceleration() {
        return this.acceleration;
    }

    getTimeDilation() {
        return this.timeDilation;
    }

    getTickFrequence() {
        return this.tickFrequency;
    }

}
